### DISCUSSION TITLE
Line rendered is off by one

### DISCUSSION THREAD

**Discussion by zslayton** [helix-editor/helix#10697] [Troubleshooting] [answered]
### Summary

When I open any file, the editor renders the file starting on line 2 instead of line 1. The cursor itself is on line 1, because moving the cursor around causes each cell to be redrawn with the correct underlying data as is shown in this recording:

https://github.com/helix-editor/helix/assets/611616/600feae5-00ce-448f-bbcf-5c4c85d35c8c

This is on mac os Sonoma. I installed helix via `brew install helix`.
```sh
❯ hx --version
helix 24.3 (2cadec0b) # The version number appears to be a typo? Brew reports it as 24.03.
```

Reinstalling the editor did not address this. Running the `:redraw` command has the same effect as opening the file afresh -- it still starts on line 2. My `~/.config/helix` directory is empty.

Please let me know what additional information I can provide.

### Reproduction Steps

 

I tried this:

1. `hx example.txt`

I expected this to happen:

The file opens starting at line 1. The contents of each cell reflect the content of the file at that position.

Instead, this happened:

The file opens at line 1, but the first line rendered is line 2. The contents of each cell are actually the corresponding cell from the line below.

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>

```
please provide a copy of `~/.cache/helix/helix.log` here if possible, you may need to redact some of the lines
```

</details>

### Platform

macOS

### Terminal Emulator

iTerm2 Build 3.4.23

### Installation Method

brew

### Helix Version

24.03

**Comment by the-mikedavis** [comment]
Can you reproduce this is any other terminal emulator? I can't reproduce this even when running iterm2

**Reply by zslayton** [reply, marked as answer]
Great question--it seems to work fine in mac OS' provided `Terminal` app.

Based on this, I was able to narrow down the differences until I found the fix. In `Terminal`, the `TERM` environment variable was set to `xterm-256color` while in `iTerm2` it was set to `ansi`. I don't know how or why it was set to that, but switching it to `xterm-256color` addressed the problem. I'd be curious to understand the difference between the two settings, but I'm just happy to have my editor back.
